File: VB/WinSolution.Module/File.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module
	Public Class File
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property
		Private _domainObjectBase As DomainObjectBase
		<Association("DomainObjectBase-Files")> _
		Public Property DomainObjectBase() As DomainObjectBase
			Get
				Return _domainObjectBase
			End Get
			Set(ByVal value As DomainObjectBase)
				SetPropertyValue("DomainObjectBase", _domainObjectBase, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/WinSolution.Module/File.cs
using DevExpress.Xpo;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module {
    public class File : BaseObject {
        public File(Session session) : base(session) { }
        private string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }
        private DomainObjectBase _domainObjectBase;
        [Association("DomainObjectBase-Files")]
        public DomainObjectBase DomainObjectBase
        {
        	get {	return _domainObjectBase; }
        	set { SetPropertyValue("DomainObjectBase", ref _domainObjectBase, value);	}
        }
    }
}
